### PR TITLE
[Task-2] Created e2e and unit tests for Count, List and Search components.

### PR DESCRIPTION
**Pull request by PetkevichVladislav**
Unit and component testing

Check that you have [jest](https://jestjs.io/docs/getting-started) and [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/) packages installed for unit and component tests.

If you haven't created all components from the previous module, finish them now. Cover the logic for all three with tests using these libraries.

⚛️ Counter

Test that component renders initial value provided in props

Test that a click event on "decrement" button decrements the displayed value

Test that a click event on "increment" button increments the displayed value

⚛️ SearchForm

Test that component renders an input with the value equal to initial value passed in props

Test that after typing to the input and a "click" event on the Submit button, the "onChange" prop is called with proper value

Test that after typing to the input and pressing Enter key, the "onChange" prop is called with proper value

⚛️ GenreSelect

Test that component renders all genres passed in props

Test that component highlights a selected genre passed in props

Test that after a click event on a genre button component calls "onChange" callback and passes correct genre in arguments

 

End-to-End Testing

Install and setup [Cypress](https://www.cypress.io/) or [WebdriverIO](https://webdriver.io/) for end-to-end tests.

Cover the logic for one of your components using the framework.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -22,6 +21,12 @@
         "web-vitals": "^2.1.4"
       },
       "devDependencies": {
+        "@testing-library/dom": "^9.0.1",
+        "@testing-library/react": "^14.0.0",
+        "@types/react-test-renderer": "^18.0.0",
+        "cypress": "^12.7.0",
+        "jest": "^27.5.1",
```
Comment (iZamza) [inline package-lock.json:28]: please add package-lock into gitignore

File: cypress/e2e/Counter.cy.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+describe('Counter', () => {
+    const getCounter = () => cy.get("p").should("have.class", "counter__count");
+    const getIncrementButton = () => cy.get("button").contains('increment');
+    const getDecrementButton = () => cy.get("button").contains('decrement');
+
+    beforeEach(() => {
+        cy.visit("/");
+    })
```
Comment (iZamza) [inline cypress/e2e/Counter.cy.ts:8]: the file name should be in lowercase

**Review by iZamza — APPROVED** [review]
Good job!

File: cypress/e2e/Counter.cy.ts (hunk 2 above)
Comment (PetkevichVladislav) [reply cypress/e2e/Counter.cy.ts:8]: File names renamed, thank you for you comment.

File: package-lock.json (hunk 1 above)
Comment (PetkevichVladislav) [reply package-lock.json:28]: package-lock added to gitignore, thank you for you comment.

**Pull request merged**